tests/regression/lp-1906821/task.yaml

```diff
@@ -0,0 +1,29 @@
+summary: verify that parallel instances of snaps with complex layout are correct
+
+description: |
+    The layout in https://bugs.launchpad.net/snapd/+bug/1906821 would be broken
+    with parallel instances, as due to a bug in sorting of mount entries, the
+    mapping of /var/snap/aws-iot-greengrass_prime to /var/snap/aws-iot-greengrass
+    was established too late. Layout entries referring to $SNAP_DATA would access
+    content from real /var/snap/aws-iot-greengrass instead of the instance one.
+
+# it's enough to run it on one system
+systems: [ubuntu-20.04-64]
+
+prepare: |
+    echo "Ensure feature flags are enabled"
+    snap set system experimental.parallel-instances=true
+
+execute: |
+    echo "Install snaps"
+    "$TESTSTOOLS"/snaps-state install-local test-snapd-complex-layout
+    "$TESTSTOOLS"/snaps-state install-local-as test-snapd-complex-layout test-snapd-complex-layout_prime
+
+    mkdir -p /var/snap/test-snapd-complex-layout/current/certs
+    mkdir -p /var/snap/test-snapd-complex-layout_prime/current/certs
+    touch /var/snap/test-snapd-complex-layout/current/certs/non-instance
+    touch /var/snap/test-snapd-complex-layout_prime/current/certs/prime-instance
+    #shellcheck disable=SC2016
+    test-snapd-complex-layout.sh -c 'test -e $SNAP/certs/non-instance'
```

Comment:
This looks good but for completion I would check that the other file is _not_ present, just in case this passes due to some weird cross-sharing.

Reply:
done